src/ui/features/turnstile/helpers.ts

```diff
@@ -0,0 +1,7 @@
+import { emitter } from 'src/ui/shared/events';
+
+export function openTurnstileWidgetIfNeeded(response: Response) {
+  if (response.headers.get('cf-mitigated') === 'challenge') {
+    emitter.emit('openTurnstile');
```

Comment:
Maybe a local emitter (created somewhere within `src/ui/features/turnstile/`) is more appropriate here, because the purpose of this emitter is not making the event globally known, but a mechanism to change the state of an existing React Component